File: 04_decorating_the_christmas_tree.py
from io import StringIO

def create_xmas_tree(height, ornament):
  buffer = StringIO()
  for i in range(1,height+1):
    buffer.write(f"{'_'*(height-i)}{ornament*(2*n-i)}{'_'*(height-i)}\n")
  
  buffer.write(f"{'_'*(height-1)}#{'_'*(height-1)}\n")
  buffer.write(f"{'_'*(height-1)}#{'_'*(height-1)}")

  return buffer.getValue().strip

File: 04_decorating_the_christmas_tree.ts
function createXmasTree(height: number, ornament: string): string {
    let tree: string[] = []
    for (let i = 1; i <= height; i++) {
        tree.push('_'.repeat(height - i) + ornament.repeat(2 * i - 1) + "_".repeat(height - i))
    }
    tree.push("_".repeat(height - 1) + "#" + "_".repeat(height - 1))
    tree.push("_".repeat(height - 1) + "#" + "_".repeat(height - 1))
    return tree.join("\n")
}
